### PR TITLE
Add 'authoritative' parameter

### LINKED ISSUE #87: Parameterise "authoritative" statement
The "authoritative" statement in dhcpd.conf should be parameterised, so it can be disabled.

Based on https://github.com/theforeman/puppet-dhcp/commit/e29568a079c93f2382927759b116330e895b058e, though that defaults to false/not authoritative.

### PR DESCRIPTION
**Pull request by domcleal**
Fixes #87

### CODE REVIEW

File: templates/dhcpd.conf-header.erb [hunk 1]
```diff
@@ -2,7 +2,9 @@
 # ----------
 # dhcpd.conf
 # ----------
+<% if @authoritative -%>
 authoritative;
+<% end -%>
```
Comment (alexjfisher) [inline templates/dhcpd.conf-header.erb]: Is it worth adding
```
<% else -%>
not authoritative;
```
??

Comment (domcleal) [inline templates/dhcpd.conf-header.erb]: Thanks, updated to include it - better to be explicit.

File: spec/classes/dhcp_spec.rb [hunk 2]
```diff
@@ -79,6 +83,19 @@
...
+
+      context 'authoritative => false' do
+        let :params do
+          default_params.merge(
+            interfaces: ['eth0'],
+            authoritative: false
+          )
+        end
+
+        it 'has no authoritative statement' do
```
Comment (alexjfisher) [inline spec/classes/dhcp_spec.rb]: typo? (or perhaps just unmodified since last commit)

Comment (domcleal) [inline spec/classes/dhcp_spec.rb]: Unmodified! Thanks, fixed.

**Pull request merged**